### ISSUE TITLE
Inaccurate Attack Path created when ObjectType specified in ACE

### ISSUE THREAD

**Issue by lbrauns** [SpecterOps/BloodHound#613]
**Description:**

In the domain i am analyzing (BHE) the EXCHANGE RECIPIENT ADMINISTRATORS are shown with a lot of WriteDacl and WriteOwner edges. Even to the Domain Root Object:

![image](https://github.com/SpecterOps/BloodHound/assets/16474792/a7300195-ba55-475b-8af4-ce2e6860e012)

The only existing ACL that could produce this edge are ACL on the domain root object where the EXCHANGE RECIPIENT ADMINISTRATORS receive ``CreateChild, DeleteChild, Self, WriteProperty, DeleteTree, ExtendedRight, Delete, WriteDacl, WriteOwner`` on the object type ``ms-Exch-Dynamic-Distribution-List``.

Full ACL:

````
InheritedObjectType   : All
ObjectDN              : DC=CHILDDOMAIN,DC=net
ObjectType            : ms-Exch-Dynamic-Distribution-List
IdentityReference     : PARENTDOMAIN\Exchange Recipient Administrators
IsInherited           : False
ActiveDirectoryRights : CreateChild, DeleteChild, Self, WriteProperty, DeleteTree, ExtendedRight, Delete, WriteDacl, WriteOwner
PropagationFlags      : None
ObjectFlags           : ObjectAceTypePresent
InheritanceFlags      : ContainerInherit
InheritanceType       : All
AccessControlType     : Allow
ObjectSID             : S-1-5-21-[REDACTED]
TargetObjectType      : DomainRoot
````

In this case the EXCHANGE RECIPIENT ADMINISTRATORS comes from a parent domain and the privileges are present on the child domain root object, this might be relevant.

This is a false positive and creates a LOT of unnecessary edges. 

**Component(s) Affected:**

- [ ] UI
- [ ] API
- [ ] Neo4j
- [ ] PostgreSQL
- [ ] Data Collector (SharpHound, AzureHound)
- [ ] Other (tooling, documentation, etc.)

**Steps to Reproduce:**

1. Create environment with parent and child domain
2. Install Exchange Organization in the parent domain in Active Directory Split Permission mode.
3. Verify EXCHANGE RECIPIENT ADMINISTRATORS have a permission on the domain root object of the child domain
4. Collect data and run analysis
5. Check outbound control of EXCHANGE RECIPIENT ADMINISTRATORS

**Expected Behavior:**

Edges should not be created as permissions on the object type ``ms-Exch-Dynamic-Distribution-List`` are not abusable.

**Actual Behavior:**

EXCHANGE RECIPIENT ADMINISTRATORS show outbound control over all objects in the domain.

**Screenshots/Code Snippets/Sample Files:**

*If applicable, add screenshots, relevant code snippets, or sample files that help illustrate the issue.*

**Environment Information:**

BloodHound: Bloodhound Enterprise

Collector: Sharphound 2.3.10.0

**Contributor Checklist:**

- [X] I have searched the issue tracker to ensure this bug hasn't been reported before or is not already being addressed.
- [X] I have provided clear steps to reproduce the issue.
- [X] I have included relevant environment information details.
- [X] I have attached necessary supporting documents.
- [X] I have checked that any JSON files I am attempting to upload to BloodHound are valid.

**Comment by JonasBK** [comment]
Hey @lbrauns,

Thanks for reporting this.

I have Exchange configured in the AD split model in my lab, but I do not have the EXCHANGE RECIPIENT ADMINISTRATORS group at all. I guess it depends on the Exchange version.

Is it possible for you to send a screenshot of the ACE(s) the group has on the domain? Potentially from ldp.exe. Then I will try to create it manually in my lab.

**Comment by JonasBK** [comment]
Oh, you added the ACE. never mind!

**Comment by JonasBK** [comment]
I just confirmed in my lab that these edges are being generated when creating the given ACE granted to a user, and the user is not able to modify the DACL or change the owner.

When I remove the object type, then I can do those things. So it seems we should confirm that object type is not set before creating WriteDacl and WriteOwner edges. Do you agree @lbrauns?

**Comment by lbrauns** [comment]
> I just confirmed in my lab that these edges are being generated when creating the given ACE granted to a user, and the user is not able to modify the DACL or change the owner.
> 
> When I remove the object type, then I can do those things. So it seems we should confirm that object type is not set before creating WriteDacl and WriteOwner edges. Do you agree @lbrauns?

I am not sure if it is sufficient to check for the presence of an object type. The GUID map of the directory contains abusable object types as well, for example ``ms-DS-Key-Credential-Link`` is in there (but is already covered by SharpHound anyways). There might be more abusable object types contained there.

**Comment by JonasBK** [comment]
Yes, you are right. We should only avoid creating WriteDacl and WriteOwner if an object type is set. Maybe also GenericAll - I need to test that. We should still create edges like WriteAccountRestrictions that depends on the object type to be set 👍

**Comment by JonasBK** [comment]
Confirmed that GenericAll is affected. 

It does not matter if the ObjectType is an attribute that exist on the object or one that does not exist. The ACE does also not work if ObjectType is set to an object class.

Will work on a fix for SharpHound.

**Comment by JonasBK** [comment]
Interesting.. If you grant GenricAll/Full Control and set ObjectType to an attribute, access right, or validated write, then you are allowed the given access against the ObjectType, but SharpHound does not create an ACE in this case. 

![Image](https://github.com/user-attachments/assets/ab7a00d8-a13c-4868-ab85-7ff4a64f67c0)
![Image](https://github.com/user-attachments/assets/61890c95-1fc2-4696-8c7a-23b48ebaca15)
![Image](https://github.com/user-attachments/assets/fab7e247-e071-4776-a419-e78b34dad453)

**Comment by JonasBK** [comment]
The last one is the same as write access to the member attribute - it seems LDP has a bug that display it as "Add/Remove self as member"


![Image](https://github.com/user-attachments/assets/6dfb6376-2ded-4877-9b3a-d8e44702d606)

**Comment by StephenHinck** [comment]
This will be fixed in our next release with https://github.com/SpecterOps/SharpHoundCommon/pull/189
